src/plugins/auto/src/plugin.cpp

```diff
@@ -674,14 +679,21 @@ void MultiDeviceInferencePlugin::RegisterPriority(const unsigned int& priority,
 
 std::string MultiDeviceInferencePlugin::GetDeviceList(const std::map<std::string, std::string>& config) const {
     std::string allDevices;
+    std::string deviceArchitecture;
     auto deviceList = GetCore()->GetAvailableDevices();
     auto deviceListConfig = config.find(ov::device::priorities.name());
     for (auto&& device : deviceList) {
         // filter out the supported devices
-        if (!_pluginConfig.isSupportedDevice(device))
+        if (device.find("GPU") != std::string::npos) {
+            deviceArchitecture = GetCore()->GetMetric(device, METRIC_KEY(DEVICE_ARCHITECTURE)).as<std::string>();
+        }
+        if (!_pluginConfig.isSupportedDevice(device, deviceArchitecture))
             continue;
         allDevices += device + ",";
     }
+    if (allDevices.empty()) {
```

Comment:
we also need to revert this logic into the end of this function as before as AUTO should throw this exception for the case `-d AUTO:-CPU,-GPU.0,-GPU.1`. Maybe we need additional test case here to check if the exception throwed as we expected in this function.